### PR TITLE
feat: validate document with regexp

### LINKED ISSUE #9: Validar CPF com regex
Usar regex para validar se um CPF é válido no lugar de diversas funções replace

### PR DESCRIPTION
**Pull request by eakeur**
This pull request intends to implement regular expressions into the document validation module. This way, CPFs validations and replacements can be more assertive.

closes #9

### CODE REVIEW

**Comment by xeniaka** [comment]
> This pull request intends to implement regular expressions into the document validation module. This way, CPFs validations and replacements can be more assertive.
> 
> closes #9

I liked the description of your changes! 
Way to go!!!! 🚀 🚀 🚀 🚀 🚀

File: app/core/types/document/document.go [hunk 1]
```diff
@@ -1,37 +1,30 @@
...
-			return errors.New("")
-		}
+func (d Document) Validate() error {
+	if len(d) != 11 {
+		return ErrInvalidDocument
 	}
-	if len(*d) != 11 {
-		return errors.New("")
+
+	if exp.Match([]byte(d)) {
```
Comment (Leanndrosouza) [inline app/core/types/document/document.go:21]: eu suspeito que aqui vai acontecer uma coisa inesperada, que é aceitar apenas caracteres que não são digitos

File: app/core/types/document/errors.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package document
+
+import "errors"
+
+var (
+
+	// ErrInvalidDocument occurs when a CPF provided by the client is invalid or contains special characters
```
Comment (xeniaka) [inline app/core/types/document/errors.go:7]: I believe the error message you provided is clear enough... the comment would not be adding anything to it :)

**Review by xeniaka — APPROVED** [review]

**Review by Leanndrosouza — APPROVED** [review]

File: app/core/types/document/document.go (hunk 1 above)
Comment (eakeur) [reply app/core/types/document/document.go:21]: Na verdade o intuito é exatamente este. Caso ele encontre qualquer caractere que não seja numeral, entra no bloco if para retornar o erro.

File: app/core/types/document/errors.go (hunk 2 above)
Comment (eakeur) [reply app/core/types/document/errors.go:7]: Thanks for your comment :) I usually write these comments in order to keep a standard among all exported properties.

**Pull request merged**